doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,350 @@
+
+.. parsed-literal::
+   :class: highlight
+
+   my_command *[options]* *<args ...>* *<build_system_args ..>*
+
+Arguments sent to the command by the build system in this way are noted in :option:`CONFIG_SB_SIGNING_COMMAND`.
+
+.. note::
+
+   Whitespace, hyphens, and other non-alphanumeric characters must be escaped appropriately when setting the string from the command line.
+   If the custom signing command uses its own options or arguments, it is recommended to define the string in a ``prj.conf`` file or Kconfig fragment to avoid tracking backslashes.
+   Like public key paths, environment variables will not be expanded when using them in a command string set from one of these files.
+
+.. _ug_bootloader_adding_immutable_mcuboot:
+
+Adding MCUboot as an immutable bootloader
+=========================================
+
+To build :doc:`mcuboot:index` with a Zephyr or |NCS| sample, enable the :option:`CONFIG_BOOTLOADER_MCUBOOT` in the application's :file:`prj.conf` file, an associated Kconfig fragment, or using the command line:
+
+.. code-block:: console
+
+   west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- -DCONFIG_BOOTLOADER_MCUBOOT=y
+
+|how_to_configure|
+Like other child images, you can assign :ref:`image-specific configurations <ug_multi_image_variables>` at build time to further customize the bootloader's functionality.
+
+.. _ug_bootloader_adding_immutable_mcuboot_keys:
+
+Adding a custom signature key file
+----------------------------------
+
+To pass the signature key file into the MCUboot image, set its ``CONFIG_BOOT_SIGNATURE_KEY_FILE`` to the selected private key file.
+You can set the option in :file:`bootloader/mcuboot/boot/zephyr/prj.conf`, an associated Kconfig fragment, or using the command line:
+
+.. tabs::
+
+   .. group-tab:: Kconfig / prj.conf
+
+      .. code-block:: console
+
+         CONFIG_BOOT_SIGNATURE_KEY_FILE="priv.pem"
+
+   .. group-tab:: Command line
+
+      .. code-block:: console
+
+         -Dmcuboot_CONFIG_BOOT_SIGNATURE_KEY_FILE=\"priv.pem\"
+
+      Escaped quotations avoid malformed-string warnings from Kconfig.
+
+The path of the key works as :ref:`described above <ug_bootloader_adding_immutable_keys>` for |NSIB|, except the "application" directory for relative pathing is considered to be :file:`bootloader/mcuboot`.
+
+See :ref:`ug_fw_update_keys` for information on how to generate custom keys for a project.
+
+It is recommended to also set the associated configuration for a key type to ensure MCUboot compiles the public key into its image correctly.
+
+.. code-block:: console
+
+   west build -b nrf52dk_nrf52840 zephyr/samples/hello_world -- \
+   -DCONFIG_BOOTLOADER_MCUBOOT=y \
+   -Dmcuboot_CONFIG_BOOT_SIGNATURE_KEY_FILE=\"../../priv-ecdsa256.pem\" \
+   -Dmcuboot_CONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256=y
+
+.. note::
+
+   Sometimes the key-type option is automatically applied based upon available hardware.
```

Comment:
I decided to remove this part. It sounded too much like advertising a feature that the user shouldn't rely upon.

This section is supposed to be recommending to always specify the key file type to avoid compilation issues, and this message is clearer without this section.